### DISCUSSION TITLE
"create from here" option

### DISCUSSION THREAD

**Discussion by mdougher** [redimp/otterwiki#113] [Ideas]
my use case with otterwiki sometimes has fairly deep nesting of pages. I'm just now transferring all of my content in and that means creating a lot of new pages.
it would be really nice if there was a way to pre-populate the "path" for the new page based on the page i have open when i clicked the 'create new page' button.

For example, lets say i have these pages:
foo
foo/bar
foo/bar/baz

i'm currently viewing page foo/bar. If i click the create new page button, if the path could pre-populate with "foo/bar/{placeholder?}" it would save a lot of typing and eliminate a lot of extra work from my typos.
Thats obviously somewhat of a trivialized case, but i hope you can see where i'm coming from if the path is a lot deeper and each folder is longer than 3 characters lol.

i don't think this should be something that _replaces_ the "Create page" button that already exists on the sidebar, but another option for it. Maybe it should be an option underneath "Create page", or something that appears in the header bar at the top of the page that shows where you are right now.

Just my thoughts on something small that would be a 'nice to have' kind of thing. I'm *really* liking otterwiki right now!

**Comment by msinfo32github** [comment]
+1 on this feature, plus also these to make this feature easier to use:

1. Easy way to move pages, e.g. move everything in one sub-directory/folder to another. 
    - e.g. move all in folder1/*pages* to folder1/subfolder1/*pages*
    - e.g. select and drag options maybe? unsure how this could be implemented.
2. Easier way to create page indexes
    - Instead of having to create a page under a sub-folder for a page index to be created, allow creating a page index and then easier to add pages under that "page index" later on

**Comment by redimp** [comment]
Sorry I missed this discussion and the ideas brought up here. 

I see the problem when creating nested pages and ran into the same issues that @mdougher describes when putting typos into the paths ...

My idea was to add a dropdown menu with all the existing subdirectories to the `Create Page` dialog, but my mockups were not satisfactory. I like the idea of setting the current directory as (optional?) prefix for the name in the `Create Page` dialog. I will think about how this can be realized, thanks for bringing this up!

**Comment by redimp** [comment]
The ideas that @msinfo32github brought up makes me think of a kind of "Content Manager", or "File Manager" where a feature to reorganize the page structure could be implemented.

Also I never noticed that you can not `Rename` an empty folder (which is displayed as `Page Index`).

Another thing to think about how this can be realized. Thanks for bringing this to my attention!

**Comment by redimp** [comment]
Hey @mdougher, what do you think about this:

https://github.com/redimp/otterwiki/assets/112056/683a6278-4778-432d-b3bd-133deabb424a

This list of directories is based upon the last visited pages, so when you hit `Create Page` from e.g. `foo/bar` you get to toggle `foo/` and `foo/bar`.

**Comment by mdougher** [comment]
@redimp - i think this would have made my use case MUCH easier! i was adding child pages off of the same base page repeatedly, so this would make sure the base page would be in that list.

**Reply by redimp** [reply]
This feature has been added in [v2.4.3](https://github.com/redimp/otterwiki/releases/tag/v2.4.3).
